File: ruby_solutions_1-10/Euler_7.rb
#!/usr/bin/env ruby


require 'timeout'


def is_prime(n)
  if n < 2 then return false end
  if n == 2 then return true end
  for i in (3..Math.sqrt(n).to_i + 1) 
    if n.modulo(i) == 0 then return false end
  end
  return true
end

def Euler_7
  prime = 3
  count = 2 
  until count == 10001
    prime = prime + 2 
    if is_prime(prime) 
      count += 1 
    end
  end
  puts "Answer: #{prime}"
end


if __FILE__ == $0
  start = Time.now
  Euler_7()
  finish = Time.now - start
  puts "Time: #{finish}"
end

File: Rust/Euler_7.rs
// What is the 10001st prime number?

use std::time::Instant;

fn is_prime(n:u64) -> bool {
    if n < 2 || n % 2 == 0 {
        return false;
    }

    let range:u64 = (n as f64).sqrt() as u64;
    for i in 3..range+1 {
        if n % i == 0 {
            return false;
        }
    }

    return true;
}

fn main() {
    let timer = Instant::now();
    let mut count:u64 = 1;
    let mut current:u64 = 2;

    let answer = loop {
        if count == 10001 {
            break current;
        }

        current += 1;
        if is_prime(current) {
            count += 1;
        }
    };

    println!("Answer: {}", answer);
    println!("Time:  {:?}", timer.elapsed());
}
